spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/LarsSGD.scala

```diff
@@ -175,6 +197,33 @@ object LarsSGD {
   }
 
 
+  /**
+   * Check if there is LarsSGD in optimMethods If so, return the weight decay of the first found
+   * LarsSGD. Else, return None
+   * @param optimMethods
+   * @tparam T
+   * @return The weight decay of the first found LarsSGD in the optimMethods.
+   *         Or None if there is not one
+   */
+  def containsLarsSGD[T](optimMethods: Map[String, OptimMethod[T]]): Option[Double] = {
```

Comment:
How about changing to below
``` scala
optimMethods.foreach({ (name, optim) =>
      optim match {
        case d: LarsSGD[T] =>
               return Some(d.weightDecay)
      }
    })
   return None
```

Reply:
The `return` in the lambda has no effect on the outer function...

Reply:
A new commit changes this line using `find` method followed by a `map` operation, which should be more concise. For your other comments, I have modified the code as requested.